dockstore-webservice/src/main/java/io/dockstore/webservice/core/LimitedSourceFileBuilder.java

```diff
@@ -92,26 +93,45 @@ public SourceFile build() {
         }
 
         private static void setContentWithLimits(SourceFile file, String content, String path) {
-            String limitedContent = content;
-            if (content != null) {
+            // Set the properties to default values.
+            file.setContent(content);
+            file.setState(SourceFile.State.COMPLETE);
+            // Check the content and override the above settings, if necessary.
+            if (content == null) {
+                file.setContent(null);
+                file.setState(SourceFile.State.STUB);
+                logContentAction(path, "stub");
+                return;
+            } else {
                 byte[] bytes = content.getBytes(StandardCharsets.UTF_8);
+                if (Bytes.indexOf(bytes, Byte.decode("0x00")) != -1) {
+                    // Postgres cannot store strings that contain "NUL" (value 0) characters.
+                    // Thus, Dockstore cannot currently store binary files.
+                    // https://www.postgresql.org/docs/current/datatype-character.html#DATATYPE-CHARACTER
+                    // https://www.ascii-code.com/character/%E2%90%80
+                    file.setContent("Dockstore does not store binary files");
```

Comment:
If I understand correctly, if we wanted to get a list of binary files or truncated files we would need to do a SQL query with this string, so should at least be a constant. 
A hex/reason code or something quick to look up may be helpful too if we end up with a bunch of these. 

e.g. "R01 - Dockstore does not store binary files", "R02 - Dockstore does not store files of this type ", etc.